lib/motions/general-motions.coffee

```diff
@@ -379,11 +379,56 @@ class MoveToMiddleOfScreen extends MoveToScreenLine
     height = lastScreenRow - firstScreenRow
     Math.floor(firstScreenRow + (height / 2))
 
+class ScrollKeepingCursor extends MoveToLine
+  previousFirstScreenRow: 0
+  currentFirstScreenRow: 0
+
+  constructor: (@editor, @vimState) ->
+    super(@editor, @vimState)
+
+  select: (count, options) ->
+    @scrollScreen(count)
+    super(count, options)
```

Comment:
@maxbrunsfeld fixed it storing the `destination` of  `setScrollTop` before calling `super` and setting the `@editor.setScrollTop` again. I know that this is not the best solution but this happen because the `Motion.select` is changing the scroll value, I tried to sove this changing the `moveSelectionLinewise` and `moveSelectionInclusively` code but I could find a better solution.